api/oapigen/types.go

```diff
@@ -8,6 +8,13 @@ import (
 	"fmt"
 )
 
+// BufferPeriod defines model for BufferPeriod.
+type BufferPeriod struct {
+	Enabled *bool   `json:"enabled,omitempty"`
```

Comment:
What's the context behind setting `omitempty` for all fields in the API response types? I find that working with a JSON API is most predictable when the fields are present with empty values, and that `omitempty` is useful for optional relationship fields.

Reply:
Or are these objects only used as request types and not related to response types?

Reply:
So the code generation will make any field that's not specifically 'required' as part of the schema a pointer. It will also omit empty pointers. It would be up to us to initialize the pointers and return them if we want them to be included in responses, otherwise you would see.

```Json
{
   "BufferPeriod"{
     "enabled": null
     . . .
  }
}
```

So omitempty in this case just stops us from returning nil values

```Go
taskRequest{
  Name:       "test-name",
  Source:     "test-source",
  Services:   &[]string{"api", "web"},
  WorkingDir: config.String("sync-tasks"),
  BufferPeriod: &oapigen.BufferPeriod{
	  Enabled: config.Bool(false),
	  Max:     config.String(""),
	  Min:     config.String(""),
  },
}
```
If I created a task with a bufferperiod like the above max and min would both be returned in the json as empty strings: `{"buffer_period":{"enabled":false,"max":"","min":""},"name":"test-name","services":["api","web"],"source":"test-source","working_dir":"sync-tasks"}`

else if I didn't initialize Min and Max:
```Go
taskRequest{
  Name:       "test-name",
  Source:     "test-source",
  Services:   &[]string{"api", "web"},
  WorkingDir: config.String("sync-tasks"),
  BufferPeriod: &oapigen.BufferPeriod{
	  Enabled: config.Bool(false),
  },
}
```

I would get: `{"buffer_period":{"enabled":false},"name":"test-name","services":["api","web"],"source":"test-source","working_dir":"sync-tasks"}`

Reply:
Per your question about response types, yes these objects would also be part of the response type. If we feel that the response types `require` min and max to be returned, we can also create different response objects with those requirements, it would just result in duplicated code.

Reply:
Thanks for the info! Cool, yeah definitely useful for request types and reflecting required values. Response types is tricky since it doesn't seem to play nice with request types with this framework. No suggestion here just yet in how to handle it, though it's not totally uncommon to have different objects for request vs response. We'll see where this takes us, happy to move forward as-is. Thanks for navigating the code-gen stuff

Reply:
No problem, happy to help